### PR TITLE
[WIP] Aes gcm bit limit validator

### PR DESCRIPTION
**Pull request by sbuss**
@alex CC @reaperhulk This is an alternative approach to #1844, which is maybe a bit better designed and more generic.

If you think this approach is better then I will loop back with some documentation and more tests.

Potential solution for #1821

### CODE REVIEW

File: src/cryptography/hazmat/backends/commoncrypto/ciphers.py [hunk 1]
```diff
@@ -151,6 +160,8 @@ def __init__(self, backend, cipher, mode, operation):
         self.authenticate_additional_data(b"")
 
     def update(self, data):
+        self._plaintext_size_validator.update(data)
+        self._plaintext_size_validator.validate()
```
Comment (public) [inline src/cryptography/hazmat/backends/commoncrypto/ciphers.py]: If we always call `validate` right after `update` why do we need that part of the API at all?

Comment (sbuss) [inline src/cryptography/hazmat/backends/commoncrypto/ciphers.py]: I didn't think it was right for `validate` to increment a counter, such that calling `validate(data)` was not idempotent. At the same time, just calling `update` doesn't imply a validation occurs. `update_and_validate` is an acceptable compromise, though.

**Comment by sbuss** [comment]
@reaperhulk I like this approach more than #1844, but I definitely understand not wanting to add a more general solution like this when the GCM mode is the only thing which currently requires these kinds of validations. If you think other modes might require similar validations in the future I think we should go with this one.

I'll add more tests and docs to whichever you think is better.

**Comment by public** [comment]
retest this please

**Comment by alex** [comment]
Fixed by #2093.

**Pull request closed without merging**